crypto/cmac/cmac.c

```diff
@@ -180,12 +186,12 @@ int CMAC_Update(CMAC_CTX *ctx, const void *in, size_t dlen)
             return 1;
         data += nleft;
         /* Else not final block so encrypt it */
-        if (!EVP_Cipher(ctx->cctx, ctx->tbl, ctx->last_block, bl))
+        if (EVP_Cipher(ctx->cctx, ctx->tbl, ctx->last_block, bl) <= 0)
             return 0;
     }
     /* Encrypt all but one of the complete blocks left */
     while (dlen > (size_t)bl) {
-        if (!EVP_Cipher(ctx->cctx, ctx->tbl, data, bl))
+        if (EVP_Cipher(ctx->cctx, ctx->tbl, data, bl) <= 0)
```

Comment:
Is EVP_Cipher ever expeced to return < 0 ?

Reply:
Yes it is documented to do so in some situations (and indeed does do it in the issue that sparked this PR in the first place #11957)

Reply:
In this test case the return value of EVP_Cipher was 0 in 1.1.1
and -1 in master, which one was right ?
And where is this documented?

Reply:
I'll investigate the discrepancy between 1.1.1 and master.

The `EVP_Cipher` man page says this:

````
EVP_Cipher() returns the amount of encrypted / decrypted bytes, or -1
on failure, if the flag B<EVP_CIPH_FLAG_CUSTOM_CIPHER> is set for the
cipher.  EVP_Cipher() returns 1 on success or 0 on failure, if the flag
B<EVP_CIPH_FLAG_CUSTOM_CIPHER> is not set for the cipher.
````

Reply:
> In this test case the return value of EVP_Cipher was 0 in 1.1.1
and -1 in master, which one was right ?

It turns out they are both correct behaviour. In master the provided xts implementation has the EVP_CIPH_FLAG_CUSTOM_CIPHER flag defined against it (due to f7397f0d58). But in 1.1.1 this is not the case. The presence or absence of this flag changes the return value semantics of the EVP_Cipher() function (as I noted above). Therefore both master and 1.1.1 are behaving in accordance with the documentation.